tornado/iostream.py

```diff
@@ -1251,6 +1272,17 @@ def reading(self):
     def writing(self):
         return self._handshake_writing or super(SSLIOStream, self).writing()
 
+    def _got_empty_write(self, size):
+        # With OpenSSL, if we couldn't write the entire buffer,
+        # the very same string object must be used on the
+        # next call to send.  Therefore we suppress
+        # merging the write buffer after an incomplete send.
+        # A cleaner solution would be to set
+        # SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER, but this is
+        # not yet accessible from python
+        # (http://bugs.python.org/issue8240)
```

Comment:
By the way, do you know how tolerant OpenSSL is with that option?
I still had "bad write retry" errors on 3.5 before fixing the freeze logic:
https://travis-ci.org/tornadoweb/tornado/jobs/172421214#L1284